### PR TITLE
Document workflow extensions

### PR DESCRIPTION
**Pull request by stefannica**
Create a structure for more elaborated documentation covering the
available FuseML workflow extensions and pointing to the official
FuseML documentation for additional details.

### CODE REVIEW

File: images/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# FuseML Workflow Extensions
+
+This folder contains a set of container images implementing various FuseML workflow steps maintained by the FuseML team:
+
+- [MLFlow builder](builders/mlflow/README.md) - builds python runtime environment container images for codesets that are structured according to the [MLFlow Project format](https://www.mlflow.org/docs/latest/projects.html).
+- [KServe predictor](inference-services/kfserving/README.md) - deploys models using the [KServe inference platform](https://kserve.github.io/website/).
+- [Seldon Core predictor](inference-services/seldon-core/README.md) - deploys ML models using the [Seldon Core MLOps platform](https://kserve.github.io/website/).
```
Comment (jsuchome) [inline images/README.md:7]: Wrong URL for seldon-core...

File: images/builders/mlflow/README.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+      - conda-forge
+    dependencies:
+      - python=3.6
+      - pip
+      - pip:
+        - mlflow
+        - tensorflow==2.0.0
+    ```
+
+The MLflow builder has a single output: the container registry repository and tag where the built MLflow environment container image is stored. This output can be used in subsequent workflow steps to run the MLflow code from the same codeset as the one used as input.
```
Comment (jsuchome) [inline images/builders/mlflow/README.md:38]: _a/the_ tag I assume ... although that sentence reads quite strange: "the registry and the tag where ... image is stored"

File: images/builders/mlflow/README.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+- `miniconda_version` - the version (tag) of the `continuumio/miniconda3` base container image to use when building MLflow environment container images based on conda (i.e. when a `conda.yaml` file is included in the codeset). If not specified, the builder will default to `4.10.3`.
+- `base_image` - the base image to use when building MLflow environment container images based on pip (i.e. when a pip `requirements.txt` file is included in the codeset). If not specified, the builder will default to `python:3.6.13`.
+- `verbose` - set to `true` to enable verbose logging in the builder workflow step (default is `false`).
+- `compressed_caching` - use this parameter to control the compressed caching feature used by [Kaniko](https://github.com/GoogleContainerTools/kaniko) when building container images. Compressed caching is set to false by default to reduce memory usage with larger images and avoid OOM problems. Set this flag to true to enable compressed caching and reduce the image build time.
+
+
+## Using the MLflow Runtime Environment Step
```
Comment (jsuchome) [inline images/builders/mlflow/README.md:80]: While "MLflow Runtime Environment Step" is a great title, it's only at the very end of this section that you mention it as a "trainer" step. I think the actual work that this step is doing (and it _is_ calling `mlflow train` after all) should be mentioned right at the top.

File: images/README.md (hunk 1 above)
Comment (stefannica) [reply images/README.md:7]: fixed

File: images/builders/mlflow/README.md (hunk 2 above)
Comment (stefannica) [reply images/builders/mlflow/README.md]: corrected

File: images/builders/mlflow/README.md (hunk 3 above)
Comment (stefannica) [reply images/builders/mlflow/README.md]: I kept the current title but I added a note mentioning the that most common use-case for the environment step is to train ML models and that's why it's also being referred to as a "trainer" step

**Review by jsuchome — APPROVED** [review]

**Pull request merged**